### ISSUE TITLE
Incorrect target range for both AC and DC for new Kia EV6

### ISSUE THREAD

**Issue by dg362** [Hyundai-Kia-Connect/kia_uvo#959]
Kia EV6 GT Line S, RWD, MY25 model. I'm in the UK, so in the EU region.

Version: 2.28.1

I have a Kia EV6 RWD in GT Line S trim in the UK, the new MY25 model, which has 80.0kWh usable battery capacity according to both Kia and EV-Database.

My charging limits are set at 100% for AC and 80% for DC.

Against the sensors of
```
sensor.ev6_target_range_of_charge_ac
sensor.ev6_target_range_of_charge_dc
```

AC range is showing as 535 miles (861 km), DC range as 418 miles (673 km)

This is after I changed the Unit of Measure from "km" to "mi" but for completeness, I've put the km values above.

Other sensors are correct. For example with a SoC of 83.5% the EV range sensor `sensor.ev6_ev_range shows` 273.40 mi

The Kia Connect App shows the correct EV range to match the integration (in this example 273 miles).
The App is set to `mi (miles)` for Unit, `English` for Language, and `24-h format` for Time.

In the Data attributes, I see the following
```
TargetSoc:
Standard: 535
Quick: 418
```

So those values match what the integration shows.

Does this mean Kia Connect is providing bad data for the newer EV6 with the larger battery? This data doesn't match the car's WLTP of 347 miles (for the RWD model on 20" wheels).

**Comment by dg362** [comment]
Because I am in the UK, Home Assistant is set to Metric. It means it doesn't use Miles, but I don't want to change to the other method because that will mess up temperatures by changing C to F.

So I've created new template entries to resolve this, such that the underlying data in km is converted to miles.
I've used my EV sensors and naming for these but sharing if of help to anyone else.
I still have to edit the Display Precision to make this useable. But it works and mitigates the issue of HA set up for the UK but wanting to use Miles for distance. Effectively the same solution to other sensors where these are changed to `mi` in the interface.

```
  - name: EV AC Range
    state_class: measurement
    icon: mdi:ev-station
    state: "{{ ( states('sensor.ev6_target_range_of_charge_ac')|float(2) / 1.6093445|round(2) ) }}"
    unit_of_measurement: mi
    unique_id: "custom_ev_ac_range"
  - name: EV DC Range
    state_class: measurement
    icon: mdi:ev-station
    state: "{{ ( states('sensor.ev6_target_range_of_charge_dc')|float(2) / 1.6093445|round(2) ) }}"
    unit_of_measurement: mi
    unique_id: "custom_ev_dc_range"
```

**Comment by cdnninja** [comment]
Was the raw data reported in the wrong unit?   Normally you can edit the sensor right in home assistant to switch units.

**Comment by dg362** [comment]
> Was the raw data reported in the wrong unit? Normally you can edit the sensor right in home assistant to switch units.

The data was reported in km and I changed the sensor directly in HA to miles.
The problem is that the car's source data is wrong. It's as if Kia has multiplied km instead of divided it.

The new EV6 has an 84kWh battery, which is 80kWh usable. On the RWD model I have, the WLTP is 347 miles.
With AC set to 100%, HA reports an AC range of 861. This is with the sensor set to default which is `km`.
I then edited the sensor to be `mi,` and this changed to say 535 miles.
But that is wrong because it should be in the 330-350 range to align with the WLTP and whatever my current efficiency is.

So I have decided to re-apply the calculation, in effect doing this twice.
Once via editing the sensor to change from `km` to `mi` as that is what I'd normally have to do anyway.
Then again to correct the calculation overall because the source data is clearly wrong for the car.

Probably the wrong answer, but one that makes the values in the sensors look correct and align with the actual mileage and range in the vehicle.
